File: sdks/php/src/PlanOverrides/Requests/UpdatePlanOverrideDto.php
<?php

namespace NovaBilling\PlanOverrides\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class UpdatePlanOverrideDto extends JsonSerializableType
{
    /**
     * @var ?array<string> $overriddenPrices Override plan prices
     */
    #[JsonProperty('overriddenPrices'), ArrayType(['string'])]
    public ?array $overriddenPrices;

    /**
     * @var ?float $overriddenMinimumCommitment Override minimum commitment amount
     */
    #[JsonProperty('overriddenMinimumCommitment')]
    public ?float $overriddenMinimumCommitment;

    /**
     * @var ?array<string> $overriddenCharges Override charge properties
     */
    #[JsonProperty('overriddenCharges'), ArrayType(['string'])]
    public ?array $overriddenCharges;

    /**
     * @var ?array<string, mixed> $metadata Custom metadata
     */
    #[JsonProperty('metadata'), ArrayType(['string' => 'mixed'])]
    public ?array $metadata;

    /**
     * @param array{
     *   overriddenPrices?: ?array<string>,
     *   overriddenMinimumCommitment?: ?float,
     *   overriddenCharges?: ?array<string>,
     *   metadata?: ?array<string, mixed>,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->overriddenPrices = $values['overriddenPrices'] ?? null;
        $this->overriddenMinimumCommitment = $values['overriddenMinimumCommitment'] ?? null;
        $this->overriddenCharges = $values['overriddenCharges'] ?? null;
        $this->metadata = $values['metadata'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/PlanOverrides/Requests/UpdatePlanOverrideDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record UpdatePlanOverrideDto
{
    /// <summary>
    /// Plan override ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// Override plan prices
    /// </summary>
    [JsonPropertyName("overriddenPrices")]
    public IEnumerable<string>? OverriddenPrices { get; set; }

    /// <summary>
    /// Override minimum commitment amount
    /// </summary>
    [JsonPropertyName("overriddenMinimumCommitment")]
    public double? OverriddenMinimumCommitment { get; set; }

    /// <summary>
    /// Override charge properties
    /// </summary>
    [JsonPropertyName("overriddenCharges")]
    public IEnumerable<string>? OverriddenCharges { get; set; }

    /// <summary>
    /// Custom metadata
    /// </summary>
    [JsonPropertyName("metadata")]
    public Dictionary<string, object?>? Metadata { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
